packages/runtime/src/runtimes/react/components/draft-switcher/draft-switcher.tsx

```diff
@@ -46,13 +45,11 @@ export function DraftSwitcher({ isDraft }: { isDraft: boolean }) {
             : null}
         </span>
       )}
-      <script
-        type="application/json"
-        id="makeswift-draft-info"
-        dangerouslySetInnerHTML={{
-          __html: JSON.stringify({ draft: isDraft, inBuilder: isInBuilder }),
-        }}
-      />
+      {/* Insert draft mode information into the DOM to make it easier to debug draft mode-related 
+          issues on production sites */}
+      <PageMeta
+        name="makeswift-draft-info"
+        content={JSON.stringify({ draft: isDraft, inBuilder: isInBuilder })} />
```

Comment:
Replaced `<script>` with `<meta>`. In addition to triggering the original error, the `<script>` element also contained incorrect draft mode values due to the fact React only injects `<script>` content on the first render.

Note that, on the other hand, this introduces a new dependency on Next so that the `<meta>` tag gets properly hoisted to the head in React 18. It's worth considering whether we want to require React 19 for the decoupled version of the runtime.

cc: @fikrikarim

Reply:
Thank you!